cowprotocol/cowswap
# feat(permit): modals SWAP

# Summary

Fixes #3145 

Add the new permit modal to SWAP (LIMIT will come in another PR)

https://github.com/cowprotocol/cowswap/assets/43217/38f28429-459f-41d5-9825-95ef90a159b7


It even [has cosmos](https://swap-dev-git-permit-3145modals-cowswap.vercel.app/cosmos/?fixtureId=%7B%22path%22%3A%22src%2Fcommon%2Fpure%2FPermitModal%2Findex.cosmos.tsx%22%2C%22name%22%3A%22Pending+permit+signature%22%7D)!

![image](https://github.com/cowprotocol/cowswap/assets/43217/0c6d687c-e90f-41fc-b923-783e258db6f3)

## Notes

Styles might need some work

# To Test

1. Pick a permittable token to sell, make sure there's no allowance
2. Using non-expert mode, confirm the swap
* Should show the new modal in the step 1 while the permit signature is requested in the wallet
3. Sign the approval
* Should show the new modal in the step 2 while the order signature is requested in the wallet
4. Sign the order
* Should show the regular modal
5. Place an order without permit
* Should show the old modal
6. Use a non-permit token that needs approval
* Approval button should be visible
8. Start the approval
* Should show the old modal
